napari/_qt/layer_controls/_tests/test_qt_layer_controls.py

```diff
@@ -79,3 +80,78 @@ def test_set_text_then_set_visible_updates_checkbox(
     layer.text.visible = True
 
     assert ctrl.textDispCheckBox.isChecked()
+
+
+# The following tests handle changes to the layer's visible and
```

Comment:
@aganders3 : here are the new tests with some refactoring of arrange/assert to help readability.

This should provide more coverage. I confirmed that your fixes made these tests fail when not present, and I also found some other issues on initialization of a control widget.

I could have parametrized these by visible/editable/enabled before/after to make these more compact, but felt like that made the tests unreadable and mostly not useful. So instead I captured some interesting visible/editable/enabled change combinations and named the tests accordingly.

One problem with parametrizing by layer instances is that the tests mutate those instances. To handle this I set editable/visible at the start of each test, but maybe there's a better way to do this (e.g. with a fixture)?

Reply:
Thanks - this looks good to me! I agree with not parametrizing too much. This makes it quite clear what's going on in each and I don't mind duplicate code in tests.

The mutability problem is interesting - I guess this is why the other tests here use the  `LayerTypeWithData` named tuple, but you're right it could also be done in a fixture. I don't think it matters here as I like the explicit setting of `layer.editable` and `layer.visible` at the start of each test anyway.